apps/users/models.py

```diff
@@ -145,6 +152,18 @@ def create_superuser(self, username, email, password):
         GroupUser.objects.create(user=user, group=admins)
         return user
 
+    def _generate_username(self, seed):
+        log.info('Generating username for {}'.format(seed))
+        if self.model.objects.filter(username=seed).exists():
+            username = '{seed}-{num}'.format(
+                seed=seed, num=random.randint(1000, 9999))
+            log.warning('Username taken for {} trying {}'.format(
+                seed, username))
+            return username
```

Comment:
Ok, all this makes sense. Could you please add it as a comment? :) I think it's valid concerns that we might have again in the future, it's better if we have the answers at hand!